src/bindings/tsd.ts

```diff
@@ -302,17 +302,21 @@ export class TSDBuilder extends ExportsWalker {
             sb.push(mode == Mode.EXPORT ? "<never>" : "<undefined>");
           }
         } else {
-          if (this.isPlainObject(clazz)) {
+          let isPlainObject = this.isPlainObject(clazz);
+          if (isPlainObject) {
             typeName = "__Record" + clazz.id.toString();
             sb.push(typeName);
             sb.push(mode == Mode.EXPORT ? "<never>" : "<undefined>");
-            this.deferredTypings.push(this.makeRecordType(clazz, mode));
           } else {
             typeName = "__Internref" + clazz.id.toString();
             sb.push(typeName);
-            this.deferredTypings.push(this.makeInternrefType(clazz));
           }
           seenObjectTypes.set(clazz, typeName);
```

Comment:
Another option is moving `seenObjectTypes.set(clazz, typeName)` over `if (this.isPlainObject(clazz)) {...} else {...}`